### ISSUE TITLE
Too long warm-up time for `TorchScript` models

### ISSUE THREAD

**Issue by liangs6212** [intel-analytics/BigDL#7062]
The initial loading of Torch Script models requires a long warm-up time, especially for the first and second calls.

### To Reproduce
```python
import torch
from bigdl.chronos.forecaster import TCNForecaster
from bigdl.chronos.data import get_public_dataset
from bigdl.nano.pytorch.inference import InferenceOptimizer

train, _, test = get_public_dataset("nyc_taxi")
dummy_data = torch.randn(1, 48, 1), torch.randn(1, 5, 1)
_loader = test.to_torch_data_loader(lookback=48, horizon=5)
tcn = TCNForecaster.from_tsdataset(train, past_seq_len=48, future_seq_len=5)
tcn.fit(train, epochs=5)
_jit_fp32 = InferenceOptimizer.trace(model=tcn.internal, input_sample=dummy_data, accelerator="jit", use_ipex=True, thread_num=1)

# first time:
with torch.no_grad():
    %time yhat = tuple(_jit_fp32(*i) for i in _loader)
# CPU times: user 618 ms, sys: 10.9 ms, total: 629 ms
# Wall time: 251 ms

# second time:
with torch.no_grad():
    %time yhat = tuple(_jit_fp32(*i) for i in _loader)
# CPU times: user 859 ms, sys: 1.18 ms, total: 860 ms
# Wall time: 456 ms

# other
with torch.no_grad():
    %time yhat = tuple(_jit_fp32(*i) for i in _loader)
# CPU times: user 411 ms, sys: 6.31 ms, total: 417 ms
# Wall time: 43.8 ms
```
Server: i9-7900(ubuntu-22.04 LTS)
Version: Python 3.7.13
Pytorch: 1.12.1
Lightning AI: 1.6.4

After comparing the inference times, the first two inference times are longer than the subsequent inference times.
This problem occurs whenever the user calls trace or load and is not described in the torch's documentation.
some similar issue: https://github.com/triton-inference-server/pytorch_backend/pull/24/files https://github.com/pytorch/pytorch/issues/57894

### Solution
`torch.jit.optimize_execution`(False) gets around this problem, but it is not mentioned in the documentation.
```python
with torch.no_grad():
    with torch.jit.optimized_execution(False):
        %time yhat = tuple(_jit_fp32(*i) for i in _loader)
# CPU times: user 530 ms, sys: 15.8 ms, total: 546 ms
# Wall time: 49.6 ms
```
In terms of results, `optimize_execution` also causes a small performance loss.
Do we need to fix this? It seems like all of our models have this problem, not just the autoformer? @TheaperDeng

**Comment by liangs6212** [comment]
Two possible solutions are shown below:

Chronos:
```python
def predict_with_jit():
    ...
    with torch.jit.optimized_execution(False):
        return _pytorch_fashion_inference(model, ...)
```

# Nano
```python
Add `optimized_execution` to nano `Contextmanager`.
```

**Comment by TheaperDeng** [comment]
I think we could provide a "warmed-up" model to our users once uses call `InferenceOptimizer.trace` with accelerator="jit". We do this in `InferenceOptimizer.optimize` while not in `InferenceOptimizer.trace`.

**Comment by TheaperDeng** [comment]
I am not really sure if we need to add this `torch.jit.optimized_execution(False):` in our context manager since it is really bad documented

**Comment by liangs6212** [comment]
> I think we could provide a "warmed-up" model to our users once uses call `InferenceOptimizer.trace` with accelerator="jit". We do this in `InferenceOptimizer.optimize` while not in `InferenceOptimizer.trace`.

btw, when the user calls `load`, it will change back to the "unwarmed-up" model.

**Comment by liangs6212** [comment]
> I am not really sure if we need to add this `torch.jit.optimized_execution(False):` in our context manager since it is really bad documented

What about `chronos`? I think `forecaster` needs this `torch.jit.optimized_execution(False)`.

**Comment by TheaperDeng** [comment]
> > I am not really sure if we need to add this `torch.jit.optimized_execution(False):` in our context manager since it is really bad documented
> 
> What about `chronos`? I think `forecaster` needs this `torch.jit.optimized_execution(False)`.

I think put it to chronos is very reasonable since we focus on a very specific solution(model) for each chrono forecaster
